### DISCUSSION TITLE
how to generate, sign and broadcast a tx programmatically with Go

### DISCUSSION THREAD

**Discussion by mostafamshk** [cosmos/community#22] [Chain Builders (Cosmos SDK)] [answered]
Hey guys, 
I'm building a cosmos-sdk based blockchain and at this time, i want to handle the tx generating and signing and encoding programmatically with go since it can't be done with REST apis and Cli command are not the best solution for user interaction. So i'm following [this instruction](https://docs.cosmos.network/v0.47/run-node/txs#programmatically-with-go) to do that. It shows some code that has to be implemented(creating a instance of txBuilder and etc.). But actually i dont know what should i do with this code. should I run it? or should i put the go file containing this code somewhere in my cosmos-sdk project?
i'm a bit confused . 
Can anyone tell me in easier words that what should i do with this code? 
thank you.

**Comment by facundomedica** [comment, marked as answer]
Depends on what you want to do, but in no cases I think you need to have that code in your chain binary. If you want to have an app (written in Go) that communicates with your chain, you can use that code to do it. If you want to have an **web** app you should look into CosmJS.
